src/evernodeone.c

```diff
@@ -212,100 +228,158 @@ int64_t hook(uint32_t reserved)
-                if (state_set(SBUF(state_buf), SBUF(STK_HOST_REG_FEE)) < 0)
-                    rollback(SBUF("Evernode: Could not update the state for host reg fee."), 1);
+                    host_reg_fee /= 2;
+                    UINT64_TO_BUF(state_buf, host_reg_fee);
+                    if (state_set(SBUF(state_buf), SBUF(STK_HOST_REG_FEE)) < 0)
+                        rollback(SBUF("Evernode: Could not update the state for host reg fee."), 1);
 
-                conf_max_reg *= 2;
-                UINT64_TO_BUF(state_buf, conf_max_reg);
-                if (state_set(SBUF(state_buf), SBUF(STK_MAX_REG)) < 0)
-                    rollback(SBUF("Evernode: Could not update state for max theoretical registrants."), 1);
-            }
+                    conf_max_reg *= 2;
+                    UINT64_TO_BUF(state_buf, conf_max_reg);
+                    if (state_set(SBUF(state_buf), SBUF(STK_MAX_REG)) < 0)
+                        rollback(SBUF("Evernode: Could not update state for max theoretical registrants."), 1);
+                }
 
-            accept(SBUF("Host registration successful."), 0);
+                accept(SBUF("Host registration successful."), 0);
+            }
+            else
+            {
+                // Continuation of re-registration flow (completion of an existing transfer).
+                COPY_BUF(host_addr, 0, (uint8_t *)(prev_host_addr + HOST_TOKEN_ID_OFFSET), 0, NFT_TOKEN_ID_SIZE);
+                const uint8_t *heartbeat_ptr = &prev_host_addr[HOST_HEARTBEAT_LEDGER_IDX_OFFSET];
+                INT64_TO_BUF(&host_addr[HOST_REG_LEDGER_OFFSET], INT64_FROM_BUF(heartbeat_ptr));
+                UINT64_TO_BUF(&host_addr[HOST_HEARTBEAT_LEDGER_IDX_OFFSET], UINT64_FROM_BUF(prev_host_addr + HOST_HEARTBEAT_LEDGER_IDX_OFFSET));
+                UINT64_TO_BUF(&host_addr[HOST_REG_TIMESTAMP_OFFSET], UINT64_FROM_BUF(prev_host_addr + HOST_REG_TIMESTAMP_OFFSET));
+
+                if (state_set(SBUF(host_addr), SBUF(STP_HOST_ADDR)) < 0)
+                    rollback(SBUF("Evernode: Could not set state for host_addr."), 1);
+
+                // Update previous TOKEN_ID state entry with the new attributes.
+                COPY_BUF(prev_token_id, HOST_ADDRESS_OFFSET, account_field, 0, ACCOUNT_ID_SIZE);
+                COPY_BUF_NON_CONST_LEN_GUARDM(prev_token_id, HOST_CPU_MODEL_NAME_OFFSET, cpu_model_ptr, 0, cpu_model_len, CPU_MODEl_NAME_LEN, 1, 1);
+                UINT16_TO_BUF(&prev_token_id[HOST_CPU_COUNT_OFFSET], cpu_count);
+                UINT16_TO_BUF(&prev_token_id[HOST_CPU_SPEED_OFFSET], cpu_speed);
+                UINT32_TO_BUF(&prev_token_id[HOST_CPU_MICROSEC_OFFSET], cpu_microsec);
+                UINT32_TO_BUF(&prev_token_id[HOST_RAM_MB_OFFSET], ram_mb);
+                UINT32_TO_BUF(&prev_token_id[HOST_DISK_MB_OFFSET], disk_mb);
+
+                // Set the STP_TOKEN_ID correctly.
+                TOKEN_ID_KEY((uint8_t *)(prev_host_addr + HOST_TOKEN_ID_OFFSET));
```

Comment:
This can be removed with [this comment](https://github.com/EvernodeXRPL/evernode-hook/pull/92/files#r1035584968)